content/articles/sharing-deployments-with-nixops.md

```diff
@@ -0,0 +1,198 @@
+Title: Sharing deployment state with nixops
+Date: 2015-09-09
+Short_summary: NixOps is a great tool for running nix-based servers. We show how to do that in a team.
+Category: Dev
+Authors: Tom
+
+[NixOps](https://github.com/NixOS/nixops) is a fantastic tool for
+getting reproducible servers running in no time. NixOps can deploy to
+many different targets including VirtualBox, Google's GCE and Amazon's
+cloud services AWS.
+
+NixOps has the downside of storing all its state (such as the IP
+address of a server) in a binary SQLite database. That in combination
+with some defaults such as absolute paths make it a bit cumbersome to
+share deployments. This post describes how to work around these
+issues.
+
+<!-- PELICAN_END_SUMMARY -->
+
+## Defining a server
+
+We'll define an extremely simple server that doesn't do anything other
+than warming this planet:
+
+```nix
+# network.nix
+let
+    region = "eu-west-1";
+in rec {
+    network.description = "test-network";
+    resources.ec2KeyPairs.test-keys = { inherit region; };
+
+    server = { resources, pkgs, lib, ... }: {
+        deployment.targetEnv = "ec2";
+        deployment.ec2.region = region;
+        deployment.ec2.instanceType = "t1.micro";
+        deployment.ec2.spotInstancePrice = 7;
+        deployment.ec2.keyPair = resources.ec2KeyPairs.test-keys;
+    };
+}
+```
+
+
+### Trick 1 and 2: Confining NixOps to a single directory
+
+Our first two tricks are to circumvent the default NixOps-behaviour of
+using absolute paths (by using angular brackets `<...>`), and to store
+the state for our deployment in a local file (by using `-s ...`):
+
+```
+$ NIX_PATH=${NIX_PATH}:. nixops create -s localstate.nixops -d test '<network.nix>'
+created deployment ‘1b699a42-5709-11e5-ad66-542696dce997’
+1b699a42-5709-11e5-ad66-542696dce997
```

Comment:
Yea, common unix way of writing to stderr for the log and to stdout for further processing.